lib/explorer/series.ex

```diff
@@ -1272,6 +1295,15 @@ defmodule Explorer.Series do
   def equal(%Series{dtype: :boolean} = left, right) when is_boolean(right),
     do: apply_impl(left, :eq, [right])
 
+  defp is_valid_length(left, right) do
+    case {Series.length(left), Series.length(right)} do
+      {l, r} when K.or(l == 0, r == 0) -> {false, nil}
```

Comment:
We should actually be able to compare empty with empty, no?

Reply:
I would say even anything with empty should work too. It just returns false for everything.

Reply:
Hey @josevalim, thanks for your feedback!

I should have made a disclaimer about this, I agree that _empty with empty_  and _empty with anything_ should work, the problem is that Polars doesn't seem to know how to deal with it yet.

Take a look at the returns when comparing empty Series directly with the `apply_impl` (to bypass intermediary clauses and send it directly to Polars)

**Note:** `S` is alias for `Explorer.Series`

## Empty with Empty
```elixir
iex > S.apply_impl(S.from_list([]),:eq,[S.from_list([])])     
#Explorer.Series<
  boolean[0]
  []
```  
I think that returning an empty Series for this is suboptimal, shouldn't we expect that when comparing two empties we should have a truthy value? After all, they are indeed equal.

## Empty with other lengths

```elixir
iex > S.apply_impl(S.from_list([1,2,3]),:eq,[S.from_list([])])
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
                                                                     left: `3`,
                                                                                right: `0`', /Users/aurus/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.0/src/compute/comparison/primitive.rs:18:5
                                                   ** (ErlangError) Erlang error: :nif_panicked
    (explorer 0.1.0-dev) Explorer.PolarsBackend.Native.s_eq(shape: (3,)
Series: '' [i64]
[
        1
        2
        3
], shape: (0,)
Series: '' [f64]
[
])
    (explorer 0.1.0-dev) lib/explorer/polars_backend/shared.ex:14: Explorer.PolarsBackend.Shared.apply_native/3
``` 
It just does not handle it well...

- - - 

Because of that, I preferred to raise a user friendly error. But I do agree that all of this should just work. Is that really a Polars issue (I could even open an issue there) or am I missing something?